mlir/lib/Quantum/Transforms/wrapper_for_wrapper.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/Support/FormatVariadic.h"
+
+#include "Quantum/Transforms/Passes.h"
+#include "Quantum/Transforms/Patterns.h"
+
+using namespace mlir;
+using namespace catalyst::quantum;
+
+namespace catalyst {
+namespace quantum {
+
+static size_t countCallsites(LLVM::LLVMFuncOp op)
+{
+    size_t count = 0;
+    op.walk([&](LLVM::CallOp callOp) { count++; });
+    return count;
+}
+
+static bool callsiteHasAttribute(LLVM::LLVMFuncOp op)
+{
+    bool retval = false;
+    ModuleOp moduleOp = cast<ModuleOp>(op->getParentOp());
+    if (!moduleOp)
+        return retval;
+    op.walk([&](LLVM::CallOp callOp) {
+        auto callee = callOp.getCallee();
+        if (callee) {
+            auto stringRef = callee.value().data();
+            Operation *fnDecl = moduleOp.lookupSymbol<LLVM::LLVMFuncOp>(stringRef);
+            retval = (bool)(fnDecl->getAttrOfType<UnitAttr>(
+                LLVM::LLVMDialect::getEmitCWrapperAttrName()));
+        }
+    });
```

Comment:
What if there are multiple call ops, would the return value just be the last one encountered?